Consul on ECS compatibility matrix docs update (#13060)
hashicorp/consul: website/content/docs/ecs/compatibility.mdx

---
layout: docs
page_title: Compatibility Matrix
description: Compatibility Matrix for Consul ECS
---

# Compatibility Matrix for Consul on ECS

For every release of Consul on ECS, `consul-ecs` binary and `consul-ecs` terraform module is updated. The terraform module and binary are tightly coupled. Ex: consul-ecs binary 0.4.1 must use consul-ecs terraform 0.4.1.

## Supported Consul versions

### Version 0.2.0 and above

| Consul Version | Compatible consul-ecs Versions  |
| -------------- | ------------------------------- |
| 1.10.x         | 0.2.x                           |

> | 1.11.x         | 0.3.0, 0.4.x                    |

Reviewer: What about Consul version 1.12.x (now released)?

Author: Jared, 
We have not yet qualified the released version with latest version of Consul.

The plan for now is to qualify the upcoming release of Consul on ECS (0.5) with the latest of Consul core (1.12) at that time. In the future as Consul on ECS gains more traction, we will add test framework to qualify existing releases (0.4, 0.5,..) with latest core (1.13, 1.14..). Right now focusing on feature velocity.

Reviewer: I can imagine a reader wondering: Is 1.12.x not listed because this page is just out of date? Or is it because 1.12.x actually isn't tested yet?

Would it make sense to have a row for Consul 1.12.x where the compatible consul-ecs version says "Not available yet"?

Author: That would be a good idea. Will update the PR!

> | 1.10.x         | 0.2.x                           |
> | 1.11.x         | 0.3.0, 0.4.x                    |

Reviewer 2: Suggest descending order for the versions; i.e., that we put the most recent versions of Consul/Consul on ECS at the top. I think this has a more natural feel and is consistent with how it's presented on the k8s compat matrix.

```suggestion
| 1.12.x         | Not available yet               |
| 1.11.x         | 0.3.0, 0.4.x                    |
| 1.10.x         | 0.2.x                           |
```

---
layout: docs
page_title: Compatibility Matrix
description: Compatibility Matrix for Consul ECS
---

# Compatibility Matrix for Consul on ECS

> For every release of Consul on ECS, `consul-ecs` binary and `consul-ecs` terraform module is updated. The terraform module and binary are tightly coupled. Ex: consul-ecs binary 0.4.1 must use consul-ecs terraform 0.4.1.

Reviewer 3: ```suggestion
For every release of Consul on ECS, the `consul-ecs` binary and `consul-ecs` Terraform module are updated. The versions of the Terraform module and binary are tightly coupled. For example, `consul-ecs` 0.4.1 binary must use the `consul-ecs` 0.4.1 Terraform module.
```
Terraform is a a proper product name and should be capitalized.

## Supported Consul versions

> ### Version 0.2.0 and above

Reviewer 3: ```suggestion
```
I don't think you need this subheading unless you were going to add a table for versions 0.1.X
